packages/salesforcedx-utils-vscode/src/helpers/paths.ts

```diff
@@ -60,28 +64,55 @@ export function getRelativeProjectPath(
   return packageDirIndex !== -1 ? fsPath.slice(packageDirIndex) : fsPath;
 }
 
-export function fileExtensionsMatch(sourceUri: vscode.Uri, targetExtension: string): boolean {
-  const extension = sourceUri.path.split('.').pop()?.toLowerCase();
+export function fileExtensionsMatch(
+  sourceUri: vscode.Uri,
+  targetExtension: string
+): boolean {
+  const extension = sourceUri.path
+    .split('.')
+    .pop()
+    ?.toLowerCase();
   return extension === targetExtension.toLowerCase();
 }
 
-function getSfdxDirectoryPath(): string {
-  return path.join(
-    getRootWorkspacePath(),
-    Global.SFDX_STATE_FOLDER
-  );
+function stateFolder(): string {
+  return hasRootWorkspace()
+    ? path.join(getRootWorkspacePath(), Global.SFDX_STATE_FOLDER)
+    : '';
+}
+
+async function metadataDirectory(): Promise<string> {
+  const username = WorkspaceContextUtil.getInstance().username;
+  return path.join(stateFolder(), 'orgs', String(username), 'metadata');
 }
 
-function getMetadataDirectoryPath(username: string): string {
-  return path.join(
-    getSfdxDirectoryPath(),
-    'orgs',
-    username,
-    'metadata'
+function apexTestResults(): string {
+  const apexTestResultsFolder = path.join(
+    stateFolder(),
+    'tools',
+    'testresults',
+    'apex'
   );
+  return apexTestResultsFolder;
+}
+
+function apexLanguageServerDatabase(): string | undefined {
+  if (!hasRootWorkspace()) {
+    return undefined;
+  }
+  const apexLangServerDbPath = path.join(stateFolder(), 'tools', 'apex.db');
```

Comment:
interesting...is this a path or a file? 
more opportunity to create sting consts (one resuable!).